Sources/SQLKit/Query/SQLCreateTrigger.swift

```diff
@@ -0,0 +1,267 @@
+                }
+
+                // This goofy looking double-if is because it could exist but be a SQLExpression
+                if self.each != nil {
+                    if let eachEnum = self.each as? SQLTriggerEach, eachEnum != .row {
+                        fatalError("CONSTRAINT triggers may only be specified FOR EACH ROW")
+                    }
+                }
+
+                statement.append("CONSTRAINT")
+            } else if self.referencedTable != nil {
+                fatalError("referencedTable may only be specified on CONSTRAINT triggers.")
+            }
+
+            statement.append("TRIGGER")
+            statement.append(self.name)
+            statement.append(self.when)
+            statement.append(self.event)
+
+            if let columns = self.columns, !columns.isEmpty {
+                if let event = self.event as? SQLTriggerEvent {
+                    guard event == .update else {
+                        fatalError("Only UPDATE triggers may specify a list of columns.")
+                    }
+
+                    if let when = self.when as? SQLTriggerWhen, when == .instead {
+                        fatalError("INSTEAD OF UPDATE triggers do not support lists of columns.")
+                    }
+                }
+
+                statement.append("OF")
+                statement.append(SQLList(columns))
+            }
+
+            statement.append("ON")
+            statement.append(self.table)
+
+            if let referencedTable = self.referencedTable {
+                statement.append("FROM")
+                statement.append(referencedTable)
+            }
+
+            if let timing = self.timing {
+                guard self.isConstraint else {
+                    fatalError("May only specify SQLTriggerTiming on CONSTRAINT triggers.")
+                }
+
+                statement.append(timing)
+            }
+
+            if let each = self.each {
+                statement.append(each)
+            } else if isConstraint {
+                statement.append(SQLTriggerEach.row)
+            }
+
+            if let condition = self.condition {
+                if let when = self.when as? SQLTriggerWhen, when == .instead {
+                    fatalError("INSTEAD OF triggers do not support WHEN conditions")
+                }
+
+                statement.append("WHEN")
+                statement.append(SQLGroupExpression(condition))
+            }
+
+            statement.append("EXECUTE PROCEDURE")
+            statement.append(self.procedure)
+        }
+    }
+
+    public func serialize(to serializer: inout SQLSerializer) {
+        switch serializer.database.dialect.name {
+        case "postgresql":
+            serializePostgreSql(&serializer)
+        case "mysql":
+            serializeMySql(&serializer)
+        default:
+            fatalError("TRIGGERS only supported in MySQL and PostgreSQL so far.")
```

Comment:
It would be nice if we chose either postgres's or mysql's to use here. The syntax for `TRIGGER` seems at least roughly similar between the variants so I think at least basic things could work.